pkg/autoupgrade/daemon.go

```diff
@@ -41,239 +31,174 @@ func Sync() {
 }
 
 type daemon struct {
-	client             kclient.Client
-	appKeysToNextCheck map[kclient.ObjectKey]nextCheckDetails
+	client           daemonClient
+	appKeysPrevCheck map[kclient.ObjectKey]time.Time
+}
+
+func newDaemon(c kclient.Client) *daemon {
+	return &daemon{
+		client:           &client{client: c},
+		appKeysPrevCheck: make(map[kclient.ObjectKey]time.Time),
+	}
 }
 
 // StartSync launches starts the daemon. It watches for new sync events coming and ensures a sync is triggered
 // periodically.
-func StartSync(ctx context.Context, client kclient.Client) error {
-	cfg, err := config.Get(ctx, client)
-	if err != nil {
-		return err
-	}
+func StartSync(ctx context.Context, client kclient.Client) {
+	d := newDaemon(client)
 
-	dd, err := time.ParseDuration(*cfg.AutoUpgradeInterval)
+	// Trigger one sync upon startup of the daemon
+	nextWait, err := d.sync(ctx, time.Now())
 	if err != nil {
-		return fmt.Errorf("failed to parse image check interval %v: %w", cfg.AutoUpgradeInterval, err)
+		logrus.Errorf("Encountered error syncing auto-upgrade apps: %v", err)
```

Comment:
Since this doesn't return an error any more, won't we start the controller with this sync never being started?

Reply:
The timer will always be started because we will use the `config.DefaultImageCheckIntervalDefault` (which is `5m`) if there is an error getting the config or parsing the check interval from the config.

I did change this slightly because your comment made me realize that we shouldn't be returning `0` if we can get the config object. We should return the `config.DefaultImageCheckIntervalDefault` (`5m`). I updated as such.

Reply:
Ah right on. Thanks for clearing that up for me.